app/models/duration.rb

```diff
@@ -1,13 +1,20 @@
 class Duration
 
   def self.duration_str(seconds)
+    seconds = seconds.to_i
     return "0:00" if seconds < 0
     m = (seconds / 60) % 60
     s = (seconds % 60).to_s.rjust(2, "0")
-    "#{m}:#{s}"
+
+    if m > 0
```

Comment:
We could probably accomplish this conditional with just right justifying the minutes, just like how the seconds are done.  It might be a bit cleaner.